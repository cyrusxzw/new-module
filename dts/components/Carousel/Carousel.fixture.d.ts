import type { CarouselProps } from './Carousel.types';
declare const CarouselFixture: CarouselProps;
export { CarouselFixture };
