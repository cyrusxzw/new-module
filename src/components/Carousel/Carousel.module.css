@import 'styles/index';

.base {
  padding: 0;

  &,
  & * {
    box-sizing: border-box;
  }
}

.slideWrapper {
  height: 100%;
}

.carousel {
  position: relative;
  z-index: var(--z-index-forground-content);
  overflow: hidden;

  picture {
    display: flex;
    height: 380px;
    padding: 0;

    @media (--viewport-md) {
      height: 380px;
    }

    @media (--viewport-lg) {
      height: 400px;
    }
  }

  :global(.slick-track) {
    display: flex;
  }

  :global(.slick-list) {
    position: relative;
    margin-top: 0;
    margin-right: 0;
    margin-bottom: 24px;
    margin-left: 0;

    @media (--viewport-md) {
      margin-right: 40px;
      margin-left: 40px;
    }

    @media (--viewport-lg) {
      margin-right: 80px;
      margin-bottom: 48px;
      margin-left: 80px;
    }
  }

  :global(.slick-slide) {
    & > div {
      height: 100%;
    }
  }
}

.item {
  display: flex;
  flex-direction: column-reverse;
  text-align: center;
  transition:
    background-color 700ms var(--easing-ease-out-cubic),
    color 700ms var(--easing-ease-out-cubic);

  a:active,
  a:hover,
  a:focus {
    background-color: var(--color-white-dark);
  }
}

.link {
  text-decoration: none;

  &:hover,
  &:focus {
    color: var(--color-black);
  }
}

.mobileCarouselIntroductionWrapper {
  margin-top: 0;
  margin-right: 20px;
  margin-bottom: 60px;
  margin-left: 20px;

  @media (--viewport-md) {
    margin-right: 40px;
    margin-bottom: 100px;
    margin-left: 40px;
  }

  p {
    max-width: 100%;
    font-size: rem(14px);
  }
}

.fullWidthSlides {
  .mobileCarouselIntroductionWrapper {
    @media (--viewport-lg) {
      margin-right: 80px;
      margin-bottom: 40px;
      margin-left: 40px;
    }
  }

  :global(.slick-list.slick-list) {
    margin-right: 0;
    margin-left: 0;
  }
}

.footer {
  position: relative;
  font-size: rem(14px);
  line-height: 1.6;

  @media (--viewport-md) {
    display: flex;
  }
}

.slideCounter {
  width: 100%;
  padding-right: 20px;
  padding-left: 20px;
  margin-bottom: 10px;

  @media (--viewport-md) {
    width: calc((100%) * 2 / 12);
    padding-right: 0;
    padding-left: 34px;
  }

  @media (--viewport-lg) {
    width: calc((100%) * 4 / 12);
    padding-left: 80px;
  }

  .dark & {
    color: var(--color-dark-decoration);
  }

  .light & {
    color: var(--color-white);
  }

  .flush.flush & {
    padding-left: 0;
  }
}

.caption {
  padding-right: 20px;
  padding-left: 20px;

  @media (--viewport-md) {
    width: calc((100%) * 10 / 12);
    padding-right: calc((100%) * 2 / 12);
    padding-left: 0;
  }

  @media (--viewport-lg) {
    width: calc((100%) * 8 / 12);
    padding-right: calc((100%) * 2 / 12);
    padding-left: calc((100%) * 1 / 12);
  }

  .dark & {
    color: var(--color-dark-decoration);
  }

  .light & {
    color: var(--color-white);
  }

  .flush.flush & {
    padding-right: 0;
  }
}

.onlyChild {
  @media (--viewport-md) {
    width: 100%;
    padding-left: 34px;
  }

  @media (--viewport-lg) {
    padding-left: 80px;
  }

  .flush.flush & {
    padding-right: 0;
    padding-left: 0;
  }
}

.loading {
  min-height: 400px;
}
