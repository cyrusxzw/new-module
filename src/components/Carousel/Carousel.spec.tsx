import React from 'react';
import { render } from '@testing-library/react';
import { CarouselFixture } from './Carousel.fixture';
import { Carousel } from './Carousel';

describe('<Carousel />', () => {
  it('should render base component correctly', () => {
    const { container } = render(
      <Carousel
        introduction={CarouselFixture.introduction}
        slides={CarouselFixture.slides}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders component correctly if no `introduction` prop is provided', () => {
    const { container } = render(<Carousel slides={CarouselFixture.slides} />);

    expect(container).toMatchSnapshot();
  });

  it('renders component correctly if no `url` member is provided on the `slides` prop', () => {
    const { container } = render(
      <Carousel
        slides={[
          {
            description: CarouselFixture.slides[0].description,
            heading: CarouselFixture.slides[0].heading,
            image: CarouselFixture.slides[0].image,
          },
        ]}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  // @TODO Address hooks + version of react + testing suite
  // it('should return `null` if the length of the `slides` prop is 0', () => {
  //   const component = shallow(<Carousel slides={[]} />);
  //   expect(component.type()).toEqual(null);
  // });
});
