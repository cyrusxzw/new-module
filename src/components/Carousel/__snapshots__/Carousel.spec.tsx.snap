// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel /> renders component correctly if no \`introduction\` prop is provided 1`] = `
<div>
  <div
    class="base dark"
  >
    <div
      class="slick-slider carousel slick-initialized"
    >
      
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="slideWrapper"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <a
                  class="base left dark item link"
                  href="/"
                  target="_self"
                  title="Link to Quisque non enim efficitur"
                >
                  <div
                    class="information dark"
                  >
                    <h5
                      class="base xXSmall dark heading dark"
                    >
                      Quisque non enim efficitur
                    </h5>
                    <div
                      class="description dark"
                    >
                      Vivamus sollicitudin ac ligula
                    </div>
                  </div>
                  <div
                    class="imageContainer"
                  >
                    <picture
                      class="base picture dark image"
                    >
                      <source
                        media="(min-width: 1025px)"
                        srcset="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                      />
                      <source
                        media="(min-width: 640px)"
                        srcset="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                      />
                      <source
                        media="(min-width: 0px)"
                        srcset="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                      />
                      <img
                        alt="Quisque non enim efficitur"
                        loading="lazy"
                      />
                    </picture>
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="1"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="slideWrapper"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <a
                  class="base left dark item link"
                  href="/"
                  target="_self"
                  title="Link to Phasellus posuere porta varius"
                >
                  <div
                    class="information dark"
                  >
                    <h5
                      class="base xXSmall dark heading dark"
                    >
                      Phasellus posuere porta varius
                    </h5>
                    <div
                      class="description dark"
                    >
                      Suspendisse elit mauris
                    </div>
                  </div>
                  <div
                    class="imageContainer"
                  >
                    <picture
                      class="base picture dark image"
                    >
                      <source
                        media="(min-width: 1025px)"
                        srcset="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                      />
                      <source
                        media="(min-width: 640px)"
                        srcset="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                      />
                      <source
                        media="(min-width: 0px)"
                        srcset="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                      />
                      <img
                        alt="Phasellus posuere porta varius"
                        loading="lazy"
                      />
                    </picture>
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="2"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="slideWrapper"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <a
                  class="base left dark item link"
                  href="/"
                  target="_self"
                  title="Link to Etiam at pharetra nisl"
                >
                  <div
                    class="information dark"
                  >
                    <h5
                      class="base xXSmall dark heading dark"
                    >
                      Etiam at pharetra nisl
                    </h5>
                    <div
                      class="description dark"
                    >
                      Aliquam metus enim, pellentesque eu
                    </div>
                  </div>
                  <div
                    class="imageContainer"
                  >
                    <picture
                      class="base picture dark image"
                    >
                      <source
                        media="(min-width: 1025px)"
                        srcset="https://www.aesop.com/medias/Aesop-Resurrection-Duet-with-Product-Large-1584x962px.png?context=bWFzdGVyfGltYWdlc3wzOTUwOTh8aW1hZ2UvcG5nfGltYWdlcy9oMWMvaDk4LzkxMTA3OTc2NDc5MDIucG5nfGYwMTE1NGQ3N2M0MDBmYTg1NzY4OTlkNTg0NTlmNmYzMWRkYmJkYzczYjFlOGEyMGMyOWVlZDRmZTRkYzE5ZjI"
                      />
                      <source
                        media="(min-width: 640px)"
                        srcset="https://www.aesop.com/medias/Aesop-Resurrection-Duet-with-Product-Medium-1238x752px.png?context=bWFzdGVyfGltYWdlc3wyNDczMDF8aW1hZ2UvcG5nfGltYWdlcy9oNjYvaDE5LzkxMTA3OTc4NzcyNzgucG5nfDk4Y2RjZjY4NDk3MGM4YzQ3YzMyMzY4NDhlYjNjNDFhNGNjNGM5MzE1NzFjNDc3YTg3NGIyNDdlYmQ1Yzc1NTA"
                      />
                      <source
                        media="(min-width: 0px)"
                        srcset="https://www.aesop.com/medias/Aesop-Resurrection-Duet-with-Product-Small-863x524px.png?context=bWFzdGVyfGltYWdlc3wxMjYxNTN8aW1hZ2UvcG5nfGltYWdlcy9oOTgvaDRmLzkxMTA3OTc5MTAwNDYucG5nfDExYTg3YzY4MmZlZTBlODU1NWJiN2IzNzBhNWYwZmExNzg2NGMzNDZhMWQyM2VjZWE4ZWVhMDljMjVmMmRiYzA"
                      />
                      <img
                        alt="Etiam at pharetra nisl"
                        loading="lazy"
                      />
                    </picture>
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="3"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="slideWrapper"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <a
                  class="base left dark item link"
                  href="/"
                  target="_self"
                  title="Link to Phasellus posuere porta varius"
                >
                  <div
                    class="information dark"
                  >
                    <h5
                      class="base xXSmall dark heading dark"
                    >
                      Phasellus posuere porta varius
                    </h5>
                    <div
                      class="description dark"
                    >
                      Suspendisse elit
                    </div>
                  </div>
                  <div
                    class="imageContainer"
                  >
                    <picture
                      class="base picture dark image"
                    >
                      <source
                        media="(min-width: 1025px)"
                        srcset="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                      />
                      <source
                        media="(min-width: 640px)"
                        srcset="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                      />
                      <source
                        media="(min-width: 0px)"
                        srcset="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                      />
                      <img
                        alt="Phasellus posuere porta varius"
                        loading="lazy"
                      />
                    </picture>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      
      
    </div>
  </div>
</div>
`;

exports[`<Carousel /> renders component correctly if no \`url\` member is provided on the \`slides\` prop 1`] = `
<div>
  <div
    class="base dark"
  >
    <div
      class="slick-slider carousel slick-initialized"
    >
      
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="slideWrapper"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <div
                  class="item"
                >
                  <div
                    class="information dark"
                  >
                    <h5
                      class="base xXSmall dark heading dark"
                    >
                      Quisque non enim efficitur
                    </h5>
                    <div
                      class="description dark"
                    >
                      Vivamus sollicitudin ac ligula
                    </div>
                  </div>
                  <div
                    class="imageContainer"
                  >
                    <picture
                      class="base picture dark image"
                    >
                      <source
                        media="(min-width: 1025px)"
                        srcset="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                      />
                      <source
                        media="(min-width: 640px)"
                        srcset="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                      />
                      <source
                        media="(min-width: 0px)"
                        srcset="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                      />
                      <img
                        alt="Quisque non enim efficitur"
                        loading="lazy"
                      />
                    </picture>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      
      
    </div>
  </div>
</div>
`;

exports[`<Carousel /> should render base component correctly 1`] = `
<div>
  <div
    class="base dark"
  >
    <div
      class="slick-slider carousel slick-initialized"
      dir="ltr"
    >
      <button
        class="base dark base hidden"
        title="next slide"
        type="button"
      >
        <svg
          aria-labelledby="chevron-1"
          class="base chevron dark"
          data-testid="data-testid-Icon"
          focusable="false"
          height="16"
          role="img"
          style="height: 16px; width: 16px;"
          tabindex="-1"
          viewBox="0 0 50 50"
          width="16"
        >
          <title
            id="chevron-1"
          >
            chevron
          </title>
          <g>
            <polygon
              class="chevron"
              points="25,31.3 4.2,10.5 0.1,14.6 25,39.5 50,14.6 45.9,10.5"
            />
          </g>
        </svg>
      </button>
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="base dark"
              >
                <header
                  class="heading offsetHeading base"
                >
                  <h2
                    class="base flush serifFont xLarge dark heading"
                  >
                    Maecenas elementum
                  </h2>
                  <h3
                    class="base flush mediumWeightFont xXSmall dark eyebrow"
                  >
                    Aenean ac tortor consectetur
                  </h3>
                </header>
                <div
                  class="content"
                >
                  <p
                    class="base dark description hasCta"
                  >
                    Maecenas elementum, metus id facilisis placerat, turpis diam sodales velit, quis sodales ipsum nunc nec velit. Aenean ac tortor consectetur, volutpat massa sed, ultricies orci.
                  </p>
                  <div
                    class="ctaWrapper"
                  >
                    <a
                      class="base hasIcon left dark"
                      href="/"
                      target="_self"
                      title="Aenean ac tortor"
                    >
                      Aenean ac tortor
                       
                      <i
                        aria-hidden="true"
                        class="icon"
                      >
                        <svg
                          aria-labelledby="rightArrow-2"
                          class="base rightArrow dark"
                          data-testid="data-testid-Icon"
                          focusable="false"
                          height="12"
                          role="img"
                          style="height: 12px; width: 12px;"
                          tabindex="-1"
                          viewBox="0 0 50 50"
                          width="12"
                        >
                          <title
                            id="rightArrow-2"
                          >
                            rightArrow
                          </title>
                          <g>
                            <path
                              d="M30.1,5.3L50,25.1L30.1,45h-6.6l18-17.6H0v-4.8h41.5l-18-17.6h6.6V5.3z"
                            />
                          </g>
                        </svg>
                      </i>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="1"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="slideWrapper"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <a
                  class="base left dark item link"
                  href="/"
                  target="_self"
                  title="Link to Quisque non enim efficitur"
                >
                  <div
                    class="information dark"
                  >
                    <h5
                      class="base xXSmall dark heading dark"
                    >
                      Quisque non enim efficitur
                    </h5>
                    <div
                      class="description dark"
                    >
                      Vivamus sollicitudin ac ligula
                    </div>
                  </div>
                  <div
                    class="imageContainer"
                  >
                    <picture
                      class="base picture dark image"
                    >
                      <source
                        media="(min-width: 1025px)"
                        srcset="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                      />
                      <source
                        media="(min-width: 640px)"
                        srcset="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                      />
                      <source
                        media="(min-width: 0px)"
                        srcset="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                      />
                      <img
                        alt="Quisque non enim efficitur"
                        loading="lazy"
                      />
                    </picture>
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="2"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="slideWrapper"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <a
                  class="base left dark item link"
                  href="/"
                  target="_self"
                  title="Link to Phasellus posuere porta varius"
                >
                  <div
                    class="information dark"
                  >
                    <h5
                      class="base xXSmall dark heading dark"
                    >
                      Phasellus posuere porta varius
                    </h5>
                    <div
                      class="description dark"
                    >
                      Suspendisse elit mauris
                    </div>
                  </div>
                  <div
                    class="imageContainer"
                  >
                    <picture
                      class="base picture dark image"
                    >
                      <source
                        media="(min-width: 1025px)"
                        srcset="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                      />
                      <source
                        media="(min-width: 640px)"
                        srcset="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                      />
                      <source
                        media="(min-width: 0px)"
                        srcset="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                      />
                      <img
                        alt="Phasellus posuere porta varius"
                        loading="lazy"
                      />
                    </picture>
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="3"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="slideWrapper"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <a
                  class="base left dark item link"
                  href="/"
                  target="_self"
                  title="Link to Etiam at pharetra nisl"
                >
                  <div
                    class="information dark"
                  >
                    <h5
                      class="base xXSmall dark heading dark"
                    >
                      Etiam at pharetra nisl
                    </h5>
                    <div
                      class="description dark"
                    >
                      Aliquam metus enim, pellentesque eu
                    </div>
                  </div>
                  <div
                    class="imageContainer"
                  >
                    <picture
                      class="base picture dark image"
                    >
                      <source
                        media="(min-width: 1025px)"
                        srcset="https://www.aesop.com/medias/Aesop-Resurrection-Duet-with-Product-Large-1584x962px.png?context=bWFzdGVyfGltYWdlc3wzOTUwOTh8aW1hZ2UvcG5nfGltYWdlcy9oMWMvaDk4LzkxMTA3OTc2NDc5MDIucG5nfGYwMTE1NGQ3N2M0MDBmYTg1NzY4OTlkNTg0NTlmNmYzMWRkYmJkYzczYjFlOGEyMGMyOWVlZDRmZTRkYzE5ZjI"
                      />
                      <source
                        media="(min-width: 640px)"
                        srcset="https://www.aesop.com/medias/Aesop-Resurrection-Duet-with-Product-Medium-1238x752px.png?context=bWFzdGVyfGltYWdlc3wyNDczMDF8aW1hZ2UvcG5nfGltYWdlcy9oNjYvaDE5LzkxMTA3OTc4NzcyNzgucG5nfDk4Y2RjZjY4NDk3MGM4YzQ3YzMyMzY4NDhlYjNjNDFhNGNjNGM5MzE1NzFjNDc3YTg3NGIyNDdlYmQ1Yzc1NTA"
                      />
                      <source
                        media="(min-width: 0px)"
                        srcset="https://www.aesop.com/medias/Aesop-Resurrection-Duet-with-Product-Small-863x524px.png?context=bWFzdGVyfGltYWdlc3wxMjYxNTN8aW1hZ2UvcG5nfGltYWdlcy9oOTgvaDRmLzkxMTA3OTc5MTAwNDYucG5nfDExYTg3YzY4MmZlZTBlODU1NWJiN2IzNzBhNWYwZmExNzg2NGMzNDZhMWQyM2VjZWE4ZWVhMDljMjVmMmRiYzA"
                      />
                      <img
                        alt="Etiam at pharetra nisl"
                        loading="lazy"
                      />
                    </picture>
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="4"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="slideWrapper"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <a
                  class="base left dark item link"
                  href="/"
                  target="_self"
                  title="Link to Phasellus posuere porta varius"
                >
                  <div
                    class="information dark"
                  >
                    <h5
                      class="base xXSmall dark heading dark"
                    >
                      Phasellus posuere porta varius
                    </h5>
                    <div
                      class="description dark"
                    >
                      Suspendisse elit
                    </div>
                  </div>
                  <div
                    class="imageContainer"
                  >
                    <picture
                      class="base picture dark image"
                    >
                      <source
                        media="(min-width: 1025px)"
                        srcset="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                      />
                      <source
                        media="(min-width: 640px)"
                        srcset="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                      />
                      <source
                        media="(min-width: 0px)"
                        srcset="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                      />
                      <img
                        alt="Phasellus posuere porta varius"
                        loading="lazy"
                      />
                    </picture>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="base dark base"
        title="next slide"
        type="button"
      >
        <svg
          aria-labelledby="chevron-3"
          class="base chevron dark"
          data-testid="data-testid-Icon"
          focusable="false"
          height="16"
          role="img"
          style="height: 16px; width: 16px;"
          tabindex="-1"
          viewBox="0 0 50 50"
          width="16"
        >
          <title
            id="chevron-3"
          >
            chevron
          </title>
          <g>
            <polygon
              class="chevron"
              points="25,31.3 4.2,10.5 0.1,14.6 25,39.5 50,14.6 45.9,10.5"
            />
          </g>
        </svg>
      </button>
      <div
        class="base"
        data-testid="data-testid-Carousel-Pagination"
      >
        <div
          class="list dark"
        >
          <div
            class="bar"
            style="width: 50%; left: 0%;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
